drivers/ethernet/eth_mcux.c

```diff
@@ -913,6 +916,9 @@ static void eth_iface_init(struct net_if *iface)
 	context->iface = iface;
 
 	ethernet_init(iface);
+	net_eth_carrier_off(iface);
```

Comment:
Might be easier to just call `net_if_flag_set(iface, NET_IF_NO_AUTO_START);` here so that the network interface init does not take the interface up in the first place.